RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -36,35 +39,59 @@
 namespace Diligent
 {
 
+namespace
+{
+
 struct BytecodeDumper
 {
     class WorkingDirectory
     {
     public:
-        WorkingDirectory(const Char* Path) :
-            m_CurrentPath{Path} {}
-
+        WorkingDirectory(const Char* Path)
+        {
+            FileSystem::SplitPathList(
+                Path, [this](const char* Data, size_t Size) -> bool {
+                    const auto Str = String{Data, Size};
+                    Push(Str.c_str());
+                    return true;
+                },
+                FileSystem::GetSlashSymbol());
+        }
 
         void Push(const Char* Path)
         {
-            m_CurrentPath = m_CurrentPath + FileSystem::GetSlashSymbol() + Path;
+            const auto CombinedPath = CombinePath(Path);
+            if (!FileSystem::PathExists(CombinedPath.c_str()))
+                FileSystem::CreateDirectory(CombinedPath.c_str());
 
-            if (!FileSystem::PathExists(m_CurrentPath.c_str()))
-                FileSystem::CreateDirectory(m_CurrentPath.c_str());
+            m_Paths.push_back(Path);
         }
 
         void Pop()
         {
-            m_CurrentPath = m_CurrentPath.substr(0, m_CurrentPath.find_last_of((FileSystem::GetSlashSymbol())));
+            m_Paths.pop_back();
         }
 
         String ComputePathFor(const Char* FileName) const
         {
-            return m_CurrentPath + FileSystem::GetSlashSymbol() + FileName;
+            return CombinePath(FileName);
         }
 
     private:
-        String m_CurrentPath;
+        String CombinePath(const Char* Path) const
+        {
+            std::stringstream  Stream;
+            std::deque<String> Queue(m_Paths);
```

Comment:
No need to copy the deque!
You can iterate it like any other container: `for (const auto& Path : m_Paths)`